### DISCUSSION TITLE
Proposal: introduce "suggested"

### DISCUSSION THREAD

**Discussion by Andarist** [json-schema-org/community#246] [Specification]
At the moment "default" is used by IDE autocompletions (it's even labeled~ in the VS Code with "A default value. Used by suggestions.", I couldn't quickly find where the text of this tooltip comes from though).

However, there are times when the better value should be preferred by suggestions. A default can't always be immediately changed because of compatibility reasons.

The prominent case for this is the `compilerOptions.strict` in the `tsconfig.json`: https://json.schemastore.org/tsconfig.json . The likelihood of somebody wanting to use `strict: false` is very low - it's highly recommended to use `strict: true` and I suspect that most people adding this property to their `tsconfig.json` want that value. It would be cool if the JSON schema would support this use case - changing the `default` in this schema, without changing the runtime default wouldn't be semantically correct.

**Comment by GabenGar** [comment]
But isn't `"default"` an annotation keyword? Ergo the "runtime default" is not a thing as far as json schema concerned. But it does mean `"default"` is a confusing misnomer existing for quite a while.

**Reply by Andarist** [reply]
I understand that `"default"` doesn't have to correspond to the "runtime default". But IMHO diverging those two values could only create confusion among users.

**Reply by GabenGar** [reply]
Introducing a new keyword won't help the situation either, since it will end up in both keywords being used. Also not all defaults can be expressed as a serializable value, such as `home directory` or `the value of an environment variable "Y"`.

**Comment by awwright** [comment]
If someone wants to suggest creating an object with certain properties, then that whole object should have a "default" that includes the suggested property. For example:

```javascript
{
type: "object",
default: { "strict": true }
}
```
